### PR TITLE
popover style for apps and user dropdown menus

### PR DESCRIPTION
**Pull request by jancborchardt**
An alternative version of https://github.com/owncloud/core/pull/14731 which doesn’t require us to change all app icons to dark versions.

What do you think @owncloud/designers @karlitschek?

@PVince81 I removed the animation and menuspeed because the jQuery slide function caused the menu to slide but the arrow triangle to directly appear. If we can fix that then we should add a slight animation back at least.

### CODE REVIEW

**Comment by jancborchardt** [comment]
Before:
dropdown-before
After:
dropdown-new

**Comment by MorrisJobke** [comment]
There is a little glitch in the triangle:

dropdown


Also the centered triangle looks weird for short names. Is it possible to move it to the right for the right menu?

**Comment by MorrisJobke** [comment]
This is on current Firefoy

**Comment by jancborchardt** [comment]
The reason I didn’t move the triangle to the right is that it looks off when it’s not flush with the downward pointing triangle. But to make that flush, it would have to be shifted to the right so far that it looks ridiculous. But yeah, like that it also looks weird.

Can you debug the triangle glitch you get there? Doesn’t happen for me for some reason.

**Comment by MorrisJobke** [comment]
It is related to the zoomlevel. I guess we can't do anything here. It's simply a rendering issue of firefox :(

**Comment by karlitschek** [comment]
definitely a step into the right direction in my opinion. :+1: Wan we make the dark grey background a bit lighter or darker so that it doesn´t "conflict" too much with the dark blue in the header?  Too bad that we can´t fix the firefox bug.

**Comment by DeepDiver1975** [comment]
@jancborchardt your js change breaks the unit test:

````
Expected true to equal false.
Error: Expected true to equal false.
    at /var/jenkins/workspace/pull-request-analyser-ng-simple/label/SLAVE/core/js/tests/specs/coreSpec.js:443

**Comment by jancborchardt** [comment]
@MorrisJobke I moved the triangle of the user menu to the left a bit. Is that better than before?

@karlitschek I made the background darker. :)

@DeepDiver1975 should be fixed now. I accidentally used `show` instead of `hide` in line 480 of js.js

**Comment by jancborchardt** [comment]
@owncloud-bot retest this please

**Comment by MorrisJobke** [comment]
Looks far better now :)

I tested it with Firefox, Chrome and IE9: Works nice :)

But in IE8 at least the background needs to be added(left menu is okay so far - right menu is completely unreadable
bildschirmfoto von 2015-03-19 17-42-01
):

**Comment by MorrisJobke** [comment]
Beside the IE8 bug: :+1:

**Comment by karlitschek** [comment]
Hmmm. I´m not really sure. Looks O.K. but not super awesome.

**Comment by karlitschek** [comment]
@owncloud/designers Does someone has a better idea?

**Comment by jancborchardt** [comment]
@karlitschek another option would be to make the blue in the header lighter. It's almost black anyway ...

**Comment by MorrisJobke** [comment]
@jancborchardt rebase needed

**Comment by jancborchardt** [comment]
Rebased and also added more fixes (logo smaller, put triangle in right place). Please review again @karlitschek @owncloud/designers

**Comment by karlitschek** [comment]
looks good in general. :+1:  The arrow in the apps dropdown is not correctly aligned for me :-)

**Comment by jancborchardt** [comment]
@Volker-E in this branch I also fixed the accesibility of the ownCloud logo https://github.com/owncloud/core/issues/15013. As well as the navigation menu – previously it was only tabbed into after tabbing over the search field and the user name.

**Comment by karlitschek** [comment]
one more problem. the login screen is also broken

**Comment by jancborchardt** [comment]
@karlitschek log in fixed, arrow alignment should be fixed too. Added proper animations.

I also added the same animations for the Versions and Sharing dropdown but for some reason they only work for the _closing_ animation of the Share dropdown. cc @schiesbn – something that we can fix later though.

**Comment by karlitschek** [comment]
Arrow alignment is still off :-/  
sharing opening animation doesn´t work as mentioned. maybe something for later or someone else can have a look?
Also the new file dropdown doesn´t seems to be animated. Not sure if this is intended.

**Comment by jancborchardt** [comment]
@karlitschek ok, I will come over later to check. About the new file dropdown: Having the animation there caused it to look strange, I can show you later as well.
And the sharing opening animation – yes, something for another pull request since it’s too unrelated to this one.

**Comment by MorrisJobke** [comment]
We (Tom, Robin, Joas, Frank) here all agree that this is a nice improvement :) Thanks Jan :)

:+1:

**Comment by MorrisJobke** [comment]
* [ ] Have a look at IE8: the personal menu is without background

**Comment by MorrisJobke** [comment]
@tomneedham Please review :)

**Comment by MorrisJobke** [comment]
Here is the before and after view: (cc @jospoortvliet for your report ;))

menu-old
menu-new

**Comment by MorrisJobke** [comment]
Jenkins PR: #15221

**Comment by jancborchardt** [comment]
Test passed via https://github.com/owncloud/core/pull/15233
https://ci.owncloud.org//job/pull-request-analyser-ng-simple/10889/

File: core/js/js.js [hunk 1]
```diff
@@ -1196,6 +1196,20 @@ function initCore() {
 
 	setupMainMenu();
 
+	// move triangle of apps dropdown to align with app name triangle
+	// 2 is the additional offset between the triangles
+	if($('#navigation').length) {
+		$('#header #owncloud + .menutoggle').one('click', function(){
```
Comment (MorrisJobke) [inline core/js/js.js]: @jancborchardt Shouldn't the `one()` be an `on()`?

Comment (MorrisJobke) [inline core/js/js.js]: Got it... it's a one time event. Nevermind

Comment (jancborchardt) [inline core/js/js.js]: Yup ;) sorcery learnt from our resident wizard @PVince81 :star2:

**Pull request merged**